### ISSUE TITLE
Hard to parse inline parsing error

### ISSUE THREAD

**Issue by hadley** [yihui/knitr#2173]
It's hard for me to understand this error (which I think is new in https://github.com/yihui/knitr/commit/3cbf47f31efebc28b59ac35c2ba07cbd3dd844ac)

``` r
knitr::knit(text = "`r 1 + `", quiet = TRUE)
#> Quitting from lines 1-1 (cilia-stag_reprex.spin.Rmd)
#> Error: Failed to parse the inline R code: 1 +  (Reason: <text>:2:0: unexpected end of input
#> 1: 1 + 
#>    ^)
```

<sup>Created on 2022-09-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I'd find it easier to understand if it was more like:

```R
#> Error: Failed to parse the inline R code: `1 + `
#> Reason: <text>:2:0: unexpected end of input
#> 1: 1 + 
#>    ^
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('knitr')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('yihui/knitr')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by yihui** [comment]
Thanks! The error message includes `` `r ` `` now, and the reason is written to the next line:

```
#> Error: Failed to parse the inline R code: `r 1 + `
#> Reason: <text>:2:0: unexpected end of input
#> 1: 1 + 
#>    ^
```

**Comment by hadley** [comment]
Much better, thanks!
